### PR TITLE
Tc uart

### PR DESCRIPTION
**Pull request by Sourland**

### CODE REVIEW

File: Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include "COBS.hpp"
+
+class TCHandlingTask : public Task {
+private:
+    const uint16_t delayMs = 1000;
+    const static inline uint16_t TaskStackDepth = 1000;
+    StackType_t taskStack[TaskStackDepth];
+
+    const static uint8_t tc_command_size = 64;
+    const static uint16_t dma_buffer_size = 2*tc_command_size;
```
Comment (Sourland) [inline Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp]: Move to Inc/Platform/COMMS_Definitions

File: Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    uint8_t rx_dma_buffer [dma_buffer_size];
+    uint8_t tc_command[tc_command_size];
```
Comment (Sourland) [inline Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp]: Docs 
https://github.com/PeakSat-AUTH/obc-software/wiki/SpaceDot-Coding-Guidelines

File: Core/Src/FreeRTOSTasks/TCHandlingTask.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#include "TCHandlingTask.hpp"
+
+extern UART_HandleTypeDef huart3;
```
Comment (Sourland) [inline Core/Src/FreeRTOSTasks/TCHandlingTask.cpp:3]: move to inc

File: Core/Src/FreeRTOSTasks/TCHandlingTask.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    while(true){
+        if(half_buffer){
+            // TC command in upper half of dma buffer [0...tc_command_size]
+            half_buffer = false;
+            std::copy(rx_dma_buffer, rx_dma_buffer+tc_command_size, tc_command);
+            // LOG_DEBUG << "TC_COMMAND[0]";
+            etl::string <tc_command_size> cobsDecodedMsg = COBSdecode<tc_command_size>(tc_command, tc_command_size);
+            uint8_t messageLength = cobsDecodedMsg.size();
+            uint8_t *ecssTCBytes = reinterpret_cast<uint8_t *>(cobsDecodedMsg.data());
```
Comment (Sourland) [inline Core/Src/FreeRTOSTasks/TCHandlingTask.cpp]:
```suggestion
            uint8_t messageLength = cobsDecodedMsg.size();
            
            uint8_t *ecssTCBytes = reinterpret_cast<uint8_t *>(cobsDecodedMsg.data());
```

File: Core/Src/FreeRTOSTasks/TCHandlingTask.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        if(full_buffer){
+            // TC command in upper half of dma buffer [tc_command_size...2*tc_command_size]
+            full_buffer = false;
+            std::copy(rx_dma_buffer+tc_command_size, rx_dma_buffer+(2*tc_command_size), tc_command);
+            // LOG_DEBUG << "TC_COMMAND[1]";
+            etl::string<tc_command_size> cobsDecodedMsg = COBSdecode<tc_command_size>(tc_command, tc_command_size);
+            uint8_t messageLength = cobsDecodedMsg.size();
+            uint8_t *ecssTCBytes = reinterpret_cast<uint8_t *>(cobsDecodedMsg.data());
+            auto ecssTC = MessageParser::parse(ecssTCBytes, messageLength);
```
Comment (Sourland) [inline Core/Src/FreeRTOSTasks/TCHandlingTask.cpp:49]:
```suggestion
            uint8_t *ecssTCBytes = reinterpret_cast<uint8_t *>(cobsDecodedMsg.data());
            
            auto ecssTC = MessageParser::parse(ecssTCBytes, messageLength);
```

File: Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp (hunk 2 above)
Comment (Sourland) [reply Core/Inc/FreeRTOSTasks/TCHandlingTask.hpp]: ETL vectors or arrays

File: Core/Inc/Platform/COMMS_Definitions.hpp [hunk 6]
```diff
@@ -26,4 +26,17 @@ inline const uint8_t MaxTickCountStringSize = 10;
...
 
+/**
+ * TC Command max size
+ */
+inline const uint8_t TcCommandSize = 64;
+
+/**
+ * The size of the allocated buffer for UART DMA
+ */
+
```
Comment (Sourland) [inline Core/Inc/Platform/COMMS_Definitions.hpp]:
```suggestion
```

**Review by giatats — APPROVED** [review]

**Pull request merged**